server/core/src/irods_generic_object.cpp

```diff
@@ -0,0 +1,111 @@
+// =-=-=-=-=-=-=-
+// irods includes
+#include "irods_generic_object.hpp"
+#include "irods_database_manager.hpp"
+
+namespace irods {
+
+// =-=-=-=-=-=-=-
+// public - ctor
+    generic_object::generic_object(const std::string &_type) : type_(_type) {
+
+    } // ctor
+
+// =-=-=-=-=-=-=-
+// public - cctor
+    generic_object::generic_object(
+        const generic_object& _rhs ) :
+        database_object( _rhs ), type_(_rhs.type_) {
+
+    } // cctor
+
+// =-=-=-=-=-=-=-
+// public - dtor
+    generic_object::~generic_object() {
+    } // dtor
+
+// =-=-=-=-=-=-=-
+// public - assignment operator
+    generic_object& generic_object::operator=(
+        const generic_object& ) {
+
+        return *this;
+
+    } // operator=
+
+// =-=-=-=-=-=-=-
+// public - equivalence operator
+    bool generic_object::operator==(
+        const generic_object& ) const {
+        return false;
+
+    } // operator==
+
+// =-=-=-=-=-=-=-
+// plugin resolution operation
+    error generic_object::resolve(
+        const std::string& _interface,
+        plugin_ptr&        _ptr ) {
+        // =-=-=-=-=-=-=-
+        // check the interface type and error out if it
+        // isnt a database interface
+        if ( DATABASE_INTERFACE != _interface ) {
+            std::stringstream msg;
+            msg << "generic_object does not support a [";
+            msg << _interface;
+            msg << "] plugin interface";
+            return ERROR( SYS_INVALID_INPUT_PARAM, msg.str() );
+
+        }
+
+        // =-=-=-=-=-=-=-
+        // ask the database manager for a postgres resource
```

Comment:
postgres?

Reply:
thanks for catching that. will update the commit